packages/loader/container-loader/src/container.ts

```diff
@@ -488,7 +485,7 @@ export class Container
 	}
 
 	public get resolvedUrl(): IResolvedUrl | undefined {
-		return this._resolvedUrl;
```

Comment:
I see the attraction of removing a member, but it does introduce local ambiguity as to if/when/why the resolved url might change over the lifetime of the Container.  Perhaps either keep the member (so we can explicitly see when it gets assigned) or at minimum add heavy commenting on expectations here.

Reply:
added comments. i prefer to minimize mutable state, especially when coupled.

Reply:
Seems fine - it does feel a little uncomfortable for this to shift inside the container without its involvement (and without it being a blackbox to the container), but I think it's due to the Container playing middle-man with the document service, getting in the way of the host directly consulting the document service as the source of truth for the url.  Probably not much to be done about that in the short term.

Reply:
yeah. fully agree. i don't have a clear picture of what a better pattern looks like right now, but reducing state changes between and within our objects, and the side effects of those state changes would be a noble goal. here is really about the attach state change, but since the contain instance doesn't change, and there are no other points of access we need to proxy a bunch of stuff 🤮